drivers/wifi/infineon/airoc_wifi.h

```diff
@@ -34,11 +62,25 @@ struct airoc_wifi_data {
 	uint8_t frame_buf[NET_ETH_MAX_FRAME_SIZE];
 };
 
+union airoc_wifi_bus {
+#if defined(CONFIG_AIROC_WIFI_BUS_SDIO)
```

Comment:
Use DT_ANY_INST_ON_BUS_STATUS_OKAY.
Although this is probably not a common case, it is best to assume that there are two devices connected via sdio and spi.